src/layout.rs

```diff
@@ -108,6 +143,26 @@ pub struct Layout {
 }
 
 impl Layout {
+    pub fn new(layout_path: Option<PathBuf>) -> Option<Self> {
+        layout_path.and_then(|path| {
+            let mut layout_file =
+                File::open(&path).expect(&format!("cannot find layout {}", &path.display()));
+
+            let mut layout = String::new();
+            layout_file
+                .read_to_string(&mut layout)
+                .expect(&format!("could not read layout {}", path.display()));
+            let layout: Layout = serde_yaml::from_str(&layout)
+                .expect(&format!("could not parse layout {}", path.display()));
+
+            if !validate_layout_percentage_total(&layout) {
```

Comment:
What do you think about moving this to its own `self.validate` method? I think this will be the first validation of many.